src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -770,4 +772,119 @@ public int sizeOfFile(File file) {
+         * @param size the size of this chunk in bytes
+         */
+        public ChunkCacheInspectionEntry(File file, long position, int size)
+        {
+            this.file = file;
+            this.position = position;
+            this.size = size;
+        }
+
+        /**
+         * Returns a string representation of this chunk cache entry.
+         *
+         * @return a formatted string containing the file path, position, and size
+         */
+        @Override
+        public String toString()
+        {
+            return String.format("Chunk{file='%s', pos=%d, size=%d}", file, position, size);
+        }
+    }
+
+    /**
+     * Inspects chunks in the cache by access frequency/recency.
+     * Uses a consumer pattern to avoid materializing a full list in memory.
+     *
+     * @param limit    maximum number of entries to inspect
+     * @param order    whether to inspect hottest (most used) or coldest (eviction candidates)
+     * @param consumer consumer to process each entry
+     */
+    public void inspectEntries(int limit, InspectEntriesOrder order, Consumer<ChunkCacheInspectionEntry> consumer)
+    {
+        boolean hottest = order == InspectEntriesOrder.HOTTEST;
+
+        if (!enabled)
+            throw new IllegalStateException("chunk cache not enabled");
+
+        // Eviction policy is required to determine hot/cold entries
+        // Note: In practice this will always be present due to maximumWeight() configuration,
+        // but we check explicitly to document the requirement and fail fast if cache setup changes.
+        if (synchronousCache.policy().eviction().isEmpty())
+            throw new IllegalStateException("no eviction policy configured - cannot determine hot/cold entries");
+
+        // The readerId packs multiple values into a single long: [File ID][Chunk Size][Reader Type]
+        // We need to shift right to extract just the File ID portion by discarding the lower bits
+        int shift = CHUNK_SIZE_LOG2_BITS + READER_TYPE_BITS;
+
+        var policy = synchronousCache.policy().eviction().get();
+        Map<Key, Chunk> orderedMap = hottest ? policy.hottest(limit) : policy.coldest(limit);
+
+        orderedMap.forEach((key, chunk) -> {
+            // Skip entries where the chunk was evicted but the key still exists
+            if (chunk == null)
+                return;
+
+            // Extract the file ID by shifting away the lower bits.
+            // The >>> operator does an unsigned right shift, moving the bits right and filling with zeros.
+            // For example, if readerId is [FileID:42][ChunkSize:3][ReaderType:1] and shift is 5,
+            // this operation discards the rightmost 5 bits (ChunkSize + ReaderType) leaving just FileID:42
+            long fileId = key.readerId >>> shift;
+
+            // Look up the File by searching through fileIdMap entries
+            File file = null;
+            for (Map.Entry<File, Long> entry : fileIdMap.entrySet())
```

Comment:
The overall time complexity would be: O (N * F) where N is the num of cache entries and F is num of files in cache..

Should we build a temporary `Map<Long, File>` above to cache the reversed mappings once? This will save some CPU cycles. Time complexity would be O(N + F)